### PR TITLE
Tangodict code improvements

### LINKED ISSUE #171: Code improvement opportunities
### Expected Behavior
N/A, same behavior, syntax change

### Actual Behavior
N/A, same behavior, syntax change

### Steps to Reproduce the Behavior
N/A, same behavior, syntax change

From this comment https://github.com/autolab/Tango/pull/160#issuecomment-674392496:

Most if not all of the keys() calls in the code do not require the list form of the keys (you only need to listify the keys if you either want to use [] to access elements of the key list, or if you will alter the dict while iterating over the keys). Furthermore, a more readable way to do
`if vmName in self.machines.keys():`

would be to implement `__contains__` in the Tango*Dictionary classes themselves and then call

`if vmName in self.machines:`

instead. TangoRemoteDictionary could use HEXISTS to implement `__contains__` and not have to fetch everything.

### PR DESCRIPTION
**Pull request by clairecw**
Implements __contains__ in the Tango*Dictionary classes, replacing any unnecessary calls to `.keys()`. 
Also adds tests for TangoDictionary accordingly. 

Resolves issue [171](https://github.com/autolab/Tango/issues/171).

### CODE REVIEW

**Comment by fanpu** [comment]
Hi @clairecw , thank you for your PR! It looks really good in general, and I really appreciate the fact that there are unit tests. A couple of more complicated unit tests would also be helpful. Do you have other instructions for end-to-end testing that you could share as well?

**Comment by clairecw** [comment]
Hi @fanpu, thanks so much for the feedback! Yes, I've added unit tests testing a bit more functionality for the dictionary object specifically - if you could TAL when you get the chance and see if there's anything else you had in mind that'd be awesome! 
Lmk also if you'd like to see unit tests for the other Tango objects.

As for E2E tests, I suppose there could be tests for each of the functions in TangoServer, and maybe another file testing the endpoints themselves.

**Comment by fanpu** [comment]
Great job and thanks! I've looked through the changes and tests and they look good, and I've also tested it with Autolab.

One last change before it's ready to be merged: could you use the [unittest](https://docs.python.org/3/library/unittest.html) methods for asserting the tests instead of the native Python assert? You can refer to [this file](https://github.com/autolab/Tango/blob/master/tests/testJobQueue.py) for examples.

**Comment by fanpu** [comment]
Tests for other Tango objects would also definitely be greatly appreciated, see https://github.com/autolab/Tango/issues/183 for some ideas

File: tests/testObjects.py [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+        
+        self.test_entries = {
+            "key": "value",
+            0: "0_value",
+            123: 456,
+        }
+
+    def runDictionaryTests(self):
+        test_dict = TangoDictionary("test")
+        assert test_dict.keys() == []
```
Comment (fanpu) [inline tests/testObjects.py:27]: Change to unittest assert

**Review by fanpu — CHANGES_REQUESTED** [review]

File: tests/testObjects.py (hunk 1 above)
Comment (clairecw) [reply tests/testObjects.py:27]: This makes sense! Thanks!

**Review by fanpu — APPROVED** [review]
Excellent job and thanks once again for your contributions!

**Pull request merged**